lib/subscriptionmanager.js

```diff
@@ -153,16 +159,29 @@ function pathMatcher (path) {
   return aPath => matcher.test(aPath)
 }
 
-function contextMatcher (selfContext, subscribeCommand) {
+function contextMatcher (selfContext, app, subscribeCommand, errorCallback) {
   debug('subscribeCommand:' + JSON.stringify(subscribeCommand))
-  var pattern = subscribeCommand.context.replace('.', '\\.').replace('*', '.*')
-  var matcher = new RegExp('^' + pattern + '$')
   if (subscribeCommand.context) {
-    return normalizedDeltaData =>
-      matcher.test(normalizedDeltaData.context) ||
-      ((subscribeCommand.context === 'vessels.self' ||
-        subscribeCommand.context === 'self') &&
-        normalizedDeltaData.context === selfContext)
+    if (_.isString(subscribeCommand.context)) {
+      var pattern = subscribeCommand.context
+        .replace('.', '\\.')
+        .replace('*', '.*')
+      var matcher = new RegExp('^' + pattern + '$')
+      return normalizedDeltaData =>
+        matcher.test(normalizedDeltaData.context) ||
+        ((subscribeCommand.context === 'vessels.self' ||
+          subscribeCommand.context === 'self') &&
+          normalizedDeltaData.context === selfContext)
+    } else if (_.get(subscribeCommand.context, 'relativePosition')) {
+      var relativePosition = subscribeCommand.context.relativePosition
+      if (!_.get(relativePosition, 'radius')) {
+        errorCallback('Please specify a radius for relativePosition')
+        return x => false
+      }
+      return normalizedDeltaData => {
+        return checkPosition(app, relativePosition, normalizedDeltaData)
```

Comment:
Doesn't checkPosition here get the relativePosition from the subscribeCommand, which is static and will never change? So we are updating the subscriber's position in spark.request, but that is never used in checkPosition?

Thinking this through how useful is the updating current position going to be in real life? It only has the lifetime of the ws connection and can be easily updated by hanging off the ws conenction and recconnecting every x hours. For this to be useful we would need to be dealing with orbital speeds or very small radii. What do you think? Ditch or keep?

Reply:
Yep. You’re right. I’ll ditch it